ddtrace/contrib/pytest/plugin.py

```diff
@@ -421,12 +461,18 @@ def pytest_runtest_protocol(item, nextitem):
                 _coverage_end(test_suite_span)
             test_suite_span.finish()
 
-        nextitem_pytest_package_item = _find_pytest_item(nextitem, pytest.Package)
-        if test_module_span is not None and (
-            nextitem is None or nextitem_pytest_package_item != pytest_package_item and not test_module_span.finished
-        ):
-            _mark_test_status(pytest_package_item, test_module_span)
-            test_module_span.finish()
+            if not module_is_package:
+                test_module_span.set_tag_str(test.STATUS, test_suite_span.get_tag(test.STATUS))
+                test_module_span.finish()
+            else:
+                nextitem_pytest_package_item = _find_pytest_item(nextitem, pytest.Package)
+                if test_module_span is not None and (
```

Comment:
I think we should get to a point where `test_module_span` cannot be `None` now. Something should've created it by now, no?

Reply:
Agree, also `test_suite_span` 👍